src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.appointment.Appointment;
+
+/**
+ * Deletes an appointment identified using it's displayed index from the CogniCare address book.
+ */
+public class DeleteAppointmentCommand extends Command {
+
+    public static final String COMMAND_WORD = "deleteappointment";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the appointment identified by the index number used in the displayed appointment list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_APPOINTMENT_SUCCESS = "Deleted Appointment: %1$s";
+
+    private final Index targetAppointmentIndex;
+
+    public DeleteAppointmentCommand(Index targetAppointmentIndex) {
+        this.targetAppointmentIndex = targetAppointmentIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Appointment> lastShownList = model.getFilteredAppointmentList();
+
+        if (targetAppointmentIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_APPOINTMENT_DISPLAYED_INDEX);
+        }
+
+        Appointment appointmentToDelete = lastShownList.get(targetAppointmentIndex.getZeroBased());
```

Comment:
This one i think is ok for now but I suggest that we use the appointment Id that is unique to each appointment.

Reply:
Noted. I might change it if there are errors during integration. If not will leave this for v1.3